ai/providers/vertexai.py

```diff
@@ -0,0 +1,131 @@
+    VERTEX_AI_PROVIDER = "VertexAI"
+    MODELS = {
+        "gemini-1.5-flash-001": {
+            "name": "Gemini 1.5 Flash 001",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-1.5-flash-002": {
+            "name": "Gemini 1.5 Flash 002",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-1.5-pro-002": {
+            "name": "Gemini 1.5 Pro 002",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-1.5-pro-001": {
+            "name": "Gemini 1.5 Pro 001",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-1.0-pro-002": {
+            "name": "Gemini 1.0 Pro 002",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-1.0-pro-001": {
+            "name": "Gemini 1.0 Pro 001",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": False,
+        },
+        "gemini-flash-experimental": {
+            "name": "Gemini Flash Experimental",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-pro-experimental": {
+            "name": "Gemini Pro Experimental",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+        "gemini-experimental": {
+            "name": "Gemini Experimental",
+            "provider": VERTEX_AI_PROVIDER,
+            "max_tokens": 8192,
+            "system_instruction_supported": True,
+        },
+    }
+
+    def __init__(self):
+        self.enabled = bool(os.environ.get("VERTEX_AI_PROJECT_ID", ""))
+        if self.enabled:
+            vertexai.init(
+                project=os.environ.get("VERTEX_AI_PROJECT_ID"),
+                location=os.environ.get("VERTEX_AI_LOCATION"),
+            )
+
+    def set_model(self, model_name: str):
+        if model_name not in self.MODELS.keys():
+            raise ValueError("Invalid model")
+        self.current_model = model_name
+
+    def get_models(self) -> dict:
+        if self.enabled:
+            return self.MODELS
+        else:
+            return {}
+
+    def generate_response(self, prompt: str, system_content: str) -> str:
+        system_instruction = None
+        if self.MODELS[self.current_model]["system_instruction_supported"]:
+            system_instruction = system_content
+        else:
+            prompt = system_content + "\n" + prompt
```

Comment:
Updated this to remove [`**kwargs`](https://github.com/calvingiles/bolt-python-ai-chatbot/blob/2077a67de56fd18711bf14954f5df2865f522e35/ai/providers/vertexai.py#L105) since it wasn't so clear which arguments were being updated before generation and I was finding this error with `pyright`:

```txt
Argument of type "str" cannot be assigned to parameter "tool_config" of type "ToolConfig | None" in function "__init__"
Type "str" is not assignable to type "ToolConfig | None"
"str" is not assignable to "ToolConfig"
"str" is not assignable to "None"
```

I found that these changes continue to work fine for both cases, and I'm hoping the set variables before model setup helps make future updates clear 🔭